{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "933805a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['meanfreq', 'sd', 'median', 'Q25', 'Q75', 'IQR', 'skew', 'kurt', 'sp.ent', 'sfm', 'mode', 'centroid', 'meanfun', 'minfun', 'maxfun', 'meandom', 'mindom', 'maxdom', 'dfrange', 'modindx']\n",
      "[18:31:24] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.5.1/src/learner.cc:576: \n",
      "Parameters: { \"silent\" } might not be used.\n",
      "\n",
      "  This could be a false alarm, with some parameters getting used by language bindings but\n",
      "  then being mistakenly passed down to XGBoost core, or some parameter actually being used\n",
      "  but getting flagged wrongly here. Please open an issue if you find any such cases.\n",
      "\n",
      "\n",
      "[0]\ttrain-logloss:0.68363\ttrain-log_loss:2.13645\tvalid-logloss:0.68453\tvalid-log_loss:2.74196\n",
      "[50]\ttrain-logloss:0.37123\ttrain-log_loss:0.24575\tvalid-logloss:0.40470\tvalid-log_loss:0.79078\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[100]\ttrain-logloss:0.21973\ttrain-log_loss:nan\tvalid-logloss:0.28004\tvalid-log_loss:nan\n",
      "[108]\ttrain-logloss:0.20301\ttrain-log_loss:nan\tvalid-logloss:0.26683\tvalid-log_loss:nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: divide by zero encountered in log\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\sklearn\\metrics\\_classification.py:2283: RuntimeWarning: invalid value encountered in multiply\n",
      "  loss = -(transformed_labels * np.log(y_pred)).sum(axis=1)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\xgboost\\core.py:108: UserWarning: ntree_limit is deprecated, use `iteration_range` or model slicing instead.\n",
      "  UserWarning\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3373: RuntimeWarning: Mean of empty slice.\n",
      "  out=out, **kwargs)\n",
      "C:\\Users\\86198\\anaconda3\\envs\\tensorflowcpu\\lib\\site-packages\\numpy\\core\\_methods.py:170: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:31:25] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.5.1/src/learner.cc:1264: Empty dataset at worker: 0\n",
      "the test acc is: nan\n"
     ]
    }
   ],
   "source": [
    "#-*- coding:utf-8 _*-  \n",
    "\n",
    "import xgboost as xgb\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sklearn\n",
    "import pickle\n",
    "import pprint\n",
    "\n",
    "def xgb_score(preds, dtrain):\n",
    "    labels = dtrain.get_label()\n",
    "    return 'log_loss', sklearn.metrics.log_loss(labels, preds)\n",
    "\n",
    "\n",
    "input_data = pd.read_csv(r'C:\\Users\\86198\\Desktop\\voice.csv')\n",
    "input_data = input_data.sample(frac=1) \n",
    "gender = {'male' : 0, 'female' : 1}\n",
    "input_data['sex'] = input_data['sex'].map(gender)\n",
    "cols = [c for c in input_data.columns if c not in ['sex']]\n",
    "print(cols)\n",
    "train = input_data.iloc[0 :3300]\n",
    "test = input_data.iloc[3300 : ]\n",
    "test_label = test['sex']\n",
    "test_label = np.array(test_label).reshape([-1 , 1])\n",
    "del(test['sex'])\n",
    "\n",
    "fold = 1\n",
    "for i in range(fold):\n",
    "    params = {\n",
    "        'eta': 0.01, #use 0.002\n",
    "        'max_depth': 5,\n",
    "        'objective': 'binary:logistic',\n",
    "        'eval_metric': 'logloss',\n",
    "        'lambda':0.1,\n",
    "        'gamma':0.1,\n",
    "        'seed': i,\n",
    "        'silent': True\n",
    "    }\n",
    "    x1 = train[cols][0:3000]\n",
    "    x2 = train[cols][3000:]\n",
    "    y1 = train['sex'][0:3000]\n",
    "    y2 = train['sex'][3000 : ]\n",
    "    watchlist = [(xgb.DMatrix(x1, y1), 'train'), (xgb.DMatrix(x2, y2), 'valid')]\n",
    "    model = xgb.train(params, xgb.DMatrix(x1, y1), 1500,  watchlist, feval=xgb_score, maximize=False, verbose_eval=50, early_stopping_rounds=50) #use 1500\n",
    "    if i != 0:\n",
    "        pred += model.predict(xgb.DMatrix(test[cols]), ntree_limit=model.best_ntree_limit)\n",
    "    else:\n",
    "        pred = model.predict(xgb.DMatrix(test[cols]), ntree_limit=model.best_ntree_limit)\n",
    "        \n",
    "pred /= fold\n",
    "pre_label = np.zeros([pred.shape[0], 1])\n",
    "for i in range(pred.shape[0]):\n",
    "    if pred[i] >= 0.5:\n",
    "        pre_label[i] = 1\n",
    "    else:\n",
    "        pre_label[i] = 0\n",
    "\n",
    "acc = np.mean(np.equal(pre_label, test_label).astype(np.float))\n",
    "print(\"the test acc is:\", acc)\n",
    "\n",
    "model_save = open('model.pkl', 'wb')    #保存模型\n",
    "pickle.dump(model, model_save)\n",
    "model_save.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "1b072825",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import xgboost as xgb\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sklearn\n",
    "import pickle\n",
    "import pprint\n",
    "import rpy2.robjects as robjects\n",
    "from rpy2.robjects import r, pandas2ri\n",
    "import os\n",
    "os.chdir(r'C:\\Users\\86198\\Desktop')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "d472f26f",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get feature file\n",
    "def get_feature(fname):\n",
    "    pandas2ri.activate()\n",
    "    robjects.r.source(r'C:\\Users\\86198\\Desktop\\feature_extract.R')\n",
    "    data_read = robjects.r.processFolder(fname)\n",
    "    data_read = pandas2ri.ri2py(data_read)\n",
    "    return data_read\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "de27bf62",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\86198\\Desktop\n",
      "C:\\Users\\86198\\Desktop\n",
      "[0.7233372  0.7233372  0.7233372  0.7233372  0.7233372  0.27783835\n",
      " 0.27783835 0.27783835 0.27783835 0.27783835]\n",
      "female is;[5.]\n",
      "male is:[5.]\n"
     ]
    }
   ],
   "source": [
    "if __name__ == '__main__':\n",
    "    data_list = []\n",
    "    model_save = open('model.pkl', 'rb')\n",
    "    model = pickle.load(model_save)\n",
    "    model_save.close()\n",
    "    print(os.getcwd())\n",
    "    file_name_list = os.listdir('data')\t\t\t\t#读取声音文件\n",
    "    print(os.getcwd())\n",
    "    for file_name in file_name_list:\n",
    "        data = get_feature(file_name)\n",
    "        data_list.append(data)\n",
    "    test = pd.concat(data_list)\n",
    "   \n",
    "    pred = model.predict(xgb.DMatrix(test), ntree_limit=model.best_ntree_limit)\n",
    "    print(pred)\n",
    "    pre_label = np.zeros([pred.shape[0], 1])\n",
    "    for i in range(pred.shape[0]):\n",
    "        if pred[i] >= 0.5:\n",
    "            pre_label[i] = 1\n",
    "        else:\n",
    "            pre_label[i] = 0\n",
    "    num = 0\n",
    "    tlen = len(pre_label)\n",
    "    for i in pre_label:\n",
    "        num += i\n",
    "    print('female is;'+str(num))\n",
    "    print( 'male is:'+str(tlen-num))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "488092a2",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5dd7097f",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ec152d6",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
